modules/apps/web-experience/export-import/export-import-resources-importer/src/main/java/com/liferay/exportimport/resources/importer/internal/messaging/ResourcesImporterHotDeployMessageListener.java

```diff
@@ -73,6 +78,16 @@
 
 	@Activate
 	protected void activate(final BundleContext bundleContext) {
+		_bundleContext = bundleContext;
+
+		_messageQueue = new LinkedList<>();
+
+		_serviceTracker = ServiceTrackerFactory.open(
```

Comment:
I'm still not sure how this helps to assure that all the required portlets are available?

The Portlet tracker becoming available does not prove that all the portlets are published. I fact if this is working it's only by pure luck. If something had blocked a needed portlet from being registered, like a missing dependency due to timing, the tracker would still activate here and the importer still fail with the portlet arriving a millisecond later.

So I think this change is not  good solution.